### PR TITLE
Added user location and calculate distance between bot and user

### PR DESCRIPTION
**Pull request by Saagar3690**
### **Changes**
Added Expo Permissions to get Location (for user)
Calculate the distance between each bot and the user for header stats

### **Todo**
Add sales stats (i.e items sold) to header stats, but first need to implement backend part of this

### **In Action**
Just a heads up, whenever the screen dims, but you don't see an Alert, it is my iPhone asking for location permissions. For some reason, it doesn't show up in the screen recording.

https://user-images.githubusercontent.com/30224372/105249904-b697f500-5b2d-11eb-98bc-7fb982051558.mov

https://user-images.githubusercontent.com/30224372/105249919-bdbf0300-5b2d-11eb-801d-0168fb6b9894.mov

https://user-images.githubusercontent.com/30224372/105249926-c3b4e400-5b2d-11eb-90f8-ffbb2908cc9f.mov

### **Bugs**
Sometimes the alert still shows up even though state has changed and it happens I think because the alert has dispatched already, but once the state changes, you can't cancel the alert, so I'm not exactly sure to fix it. You can see this happen when it says, "Oops, location permissions not granted", even though I've just granted it or when it shows the Alert for "permissions granted" twice.

### CODE REVIEW

**Review by Saagar3690 — COMMENTED** [review]
### **Changes**
Added itemsSold to headerStats for bot

### **Screenshots**
IMG_6030
IMG_6031

**Comment by Saagar3690** [comment]
This goes hand in hand with this pull request on backend: https://github.com/ucladevx/BruinBot-Backend/pull/57

File: src/containers/MapScreen.tsx
Comment (thilan-tran) [reply src/containers/MapScreen.tsx:160]: <p align="center">
  
</p>

Also, I found a bug where the `Location permissions granted` alert messages seem to be repeated, and also where the `
No access to permissions` error alert would show up even when location permissions were already enabled for Expo. (could be an Expo issue, not the code, but I'm not sure)

**Review by thilan-tran — CHANGES_REQUESTED** [review]
Looks good, just a bug with displaying repeated / unnecessary error messages.

**Review by Saagar3690 — COMMENTED** [review]
### **Changes**
Consolidated location permission variables into 1
Fixed bug where alerts for no permission even after getting permissions

### **Videos**

https://user-images.githubusercontent.com/30224372/106241821-85699580-61bb-11eb-993e-be7d7129da13.mov
https://user-images.githubusercontent.com/30224372/106241836-8bf80d00-61bb-11eb-9b39-0afc93ec5b98.mov

https://user-images.githubusercontent.com/30224372/106241940-bba71500-61bb-11eb-8b65-15a34bdfbbef.mov


### **Screenshots**
IMG_6046
IMG_6047

**Comment by Saagar3690** [comment]
DO NOT MERGE BEFORE CORRESPONDING BACKEND PULL REQUEST IS MERGED

**Review by thilan-tran — APPROVED** [review]
LGTM! Just one thing is that the "Location permission granted" alert shows up every time if you leave the map page and come back. (eg. going to the QR page and back through the drawer menu).

I'll leave merging to you after the backend PR goes through as well.

**Pull request merged**
